File: 0657-robot-return-to-origin/0657-robot-return-to-origin.java
class Solution {
    public boolean judgeCircle(String moves) {
        int x = 0;
        int y = 0;

        char [] arr = moves.toCharArray();


        for(int i =0; i<arr.length; i++){
            if(arr[i] =='U'){
                y +=1;
            }
            if(arr[i] == 'D'){
                y-=1;
            }

            if(arr[i]=='L'){
                x-=1;
            }

            if(arr[i] == 'R'){
                x+=1;
            }
        }


        if(x==0 && y == 0){
            return true;
        }

        return false;
    }
}

File: 0657-robot-return-to-origin/0657-robot-return-to-origin.cpp
#include <string>

class Solution {
public:
    bool judgeCircle(std::string moves) {
        int x = 0;
        int y = 0;

        for (char ch : moves) {
            if (ch == 'U') {
                y += 1;
            } else if (ch == 'D') {
                y -= 1;
            } else if (ch == 'L') {
                x -= 1;
            } else if (ch == 'R') {
                x += 1;
            }
        }

        return (x == 0 && y == 0);
    }
};
